commerce_coordinator/apps/frontend_app_ecommerce/views.py

```diff
@@ -25,6 +30,41 @@ def date_conv(dt: Union[datetime, str]) -> datetime:
         return dt
 
 
+# noinspection PyMethodMayBeStatic
+class RedirectReceiptView(APIView):
+    """Get the order history for the authenticated user."""
+    permission_classes = [LoginRedirectIfUnauthenticated]
+    throttle_classes = [UserRateThrottle]
+
+    def get(self, request):
+        """Get the order history for the authenticated user."""
+
+        # build parameters
+        params = {
+            'username': request.user.username,
+            "edx_lms_user_id": request.user.lms_user_id,
+            "order_number": request.query_params.get('order_number', None),
```

Comment:
We should check here if the order_number is None and return a 404 response. This will eventually not execute pipeline filters and would save us some processing logic.